File: ios/RelayLogger.swift
import Foundation
import os.log

enum RelayLogger {
  private static let prefix = "⏫ BackgroundLocationRelay: "
  private static let logger = Logger(
    subsystem: "com.margelo.nitro.backgroundlocationrelay",
    category: "BackgroundLocationRelay"
  )

  static func debug(_ message: String) {
    logger.debug("\(prefix)\(message, privacy: .public)")
  }

  static func info(_ message: String) {
    logger.info("\(prefix)\(message, privacy: .public)")
  }

  static func error(_ message: String) {
    logger.error("\(prefix)\(message, privacy: .public)")
  }
}

File: android/src/main/java/com/margelo/nitro/backgroundlocationrelay/RelayLogger.kt
package com.margelo.nitro.backgroundlocationrelay

import android.util.Log

object RelayLogger {
  private const val TAG = "BackgroundLocationRelay"
  private const val PREFIX = "⏫ BackgroundLocationRelay: "

  fun debug(message: String) {
    Log.d(TAG, "$PREFIX$message")
  }

  fun info(message: String) {
    Log.i(TAG, "$PREFIX$message")
  }

  fun error(message: String) {
    Log.e(TAG, "$PREFIX$message")
  }
}
